src/registration.js

```diff
@@ -33,52 +40,38 @@ export default class Registration extends Component {
     return optionList;
   }
 
-  handleChange = (event) => {
-    this.setState({ [event.target.name]: event.target.value });
-  }
-
-  handleSubmit = (context) => (event) => {
-    context.setUserState({ firstName: this.state.firstName, lastName: this.state.lastName, age: this.state.age });
-    this.setState({redirectToTest: true });
-  }
-
-  render() {
-    if (this.state.redirectToTest === true) {
-      return (
-        <Redirect to="/test" />
-      )
-    } else {
-      return (
-        <UserContext.Consumer>{context => (
-          <div>
-            <h2>Registration</h2>
-            <form onSubmit={this.handleSubmit(context)}>
-              <label>
-                First Name:
-            <input type="text" name="firstName" onChange={this.handleChange} />
-              </label>
-              <br></br>
-              <br></br>
-              <label>
-                Last Name:
-            <input type="text" name="lastName" onChange={this.handleChange} />
-              </label>
-              <br></br>
-              <br></br>
-              <label>
-                Age:
-              <select name="age" onChange={this.handleChange}>
-                  {this.selectListAge()}
-                </select>
-              </label>
-              <br></br>
-              <br></br>
-              <input type="submit" value="Submit" />
-            </form>
-          </div>)}
-
-        </UserContext.Consumer>
-      );
-    }
+  if (state.redirectToTest === true) {
+    return (
+      <Redirect to="/test" />
+    )
+  } else {
+    return (
+      <div>
+        <h2>Registration</h2>
+        <form onSubmit={handleSubmit(context)}>
```

Comment:
you don't need to send context to handle submit. when you get it from useContext() it's available for your component